src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1976,7 +1976,7 @@ void c_typecheck_baset::typecheck_typed_target_call(
     arguments.push_back(false_exprt());
 
   expr.arguments().swap(arguments);
-  typecheck_side_effect_function_call(expr);
+  expr.type() = empty_typet();
```

Comment:
This is surprising: isn't the expression a function call? If so, the type should `code_typet`.

Reply:
Aren't function call expressions typed according to the return type of the function ?  

if I declare a function `void foo(int i)` then `foo(x)` is of type `void` ?

Reply:
My apologies, I had confused the `side_effect_expr_function_callt` with a `code_function_callt`.